### DISCUSSION TITLE
questions about the gds generated by "make test"

### DISCUSSION THREAD

**Discussion by stefanottili** [The-OpenROAD-Project/OpenLane#1637] [General] [answered]
I just installed OpenLane and ran "make test".

Looking thru the directories, if found that the 3 gds written by magic are identical:
signoff/spm.gds
signoff/spm.magic.gds
final/gds/spm.gds 

signoff/spm.klayout.gds has the same geometry, but different layer mapping

a) why are there 3 redundant gds saved in the flow ?
b) wouldn't it be better to have these 4 gds gzipped ?
c) why would signoff/spm.klayout.gds not use the same layer mapping as the other gds ?
d) is there any plan to use the klayout lefdef map file for klayout ?
see: https://github.com/The-OpenROAD-Project/OpenROAD/discussions/1919
Using the map file allows to put def PIN on both metal and pin layer:datatype.
e) klayout also has "strm2gds/oas" buddy tools that can be used for def2gds/oas conversion.

**Comment by kareefardi** [comment, marked as answer]
So the idea was to create gds files using multiple signoff tools (klayout and magic) and then based on the configuration select on of them hence the copy `signoff/spm.gds` and then the final directory will always have copies from other parts of the flow. 
At the time of implementation magic didn't support `.gds.gz` (it should now) so perhaps we should move to .gds.gz
Finally, yes there is a plan to use klayout lefdef map file see: https://github.com/efabless/sky130_klayout_pdk/pull/2
As for oas that might be an enhancement we can look towards adding it
